Address safer C++ static analysis warnings in WebPage

cdumez opened the pull request:
#### 8f62a68186896830b810974a5c1267ae03127970
<pre>
Address safer C++ static analysis warnings in WebPage
<a href="https://bugs.webkit.org/show_bug.cgi?id=287760">https://bugs.webkit.org/show_bug.cgi?id=287760</a>

Reviewed by Darin Adler and Per Arne Vollan.

* Source/WebCore/bindings/js/DOMWrapperWorld.cpp:
(WebCore::mainThreadNormalWorldSingleton):
(WebCore::mainThreadNormalWorld): Deleted.
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
(WebCore::protectedMainThreadNormalWorld):
(WebCore::debuggerWorld):
(WebCore::pluginWorld):
* Source/WebCore/bindings/js/JSDOMWindowCustom.cpp:
(WebCore::mainWorldGlobalObject):
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
(WebCore::JSLazyEventListener::JSLazyEventListener):
(WebCore::JSLazyEventListener::create):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::evaluateIgnoringException):
(WebCore::ScriptController::loadModuleScript):
(WebCore::ScriptController::linkAndEvaluateModuleScript):
(WebCore::ScriptController::evaluateModule):
(WebCore::ScriptController::setEvalEnabled):
(WebCore::ScriptController::setWebAssemblyEnabled):
(WebCore::ScriptController::setRequiresTrustedTypes):
(WebCore::ScriptController::executeScriptIgnoringException):
(WebCore::ScriptController::executeJavaScriptURL):
(WebCore::ScriptController::reportExceptionFromScriptError):
(WebCore::ScriptController::registerImportMap):
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSVMClientData::getAllWorlds):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::pictureInPictureElement const):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::setAttributeEventListener):
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::ScriptExecutionContext::globalObject const):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplacementFragment::ReplacementFragment):
* Source/WebCore/html/HTMLBodyElement.cpp:
(WebCore::HTMLBodyElement::attributeChanged):
* Source/WebCore/html/HTMLFrameSetElement.cpp:
(WebCore::HTMLFrameSetElement::attributeChanged):
* Source/WebCore/inspector/InspectorController.cpp:
(WebCore::InspectorController::didClearWindowObjectInWorld):
* Source/WebCore/inspector/InspectorFrontendAPIDispatcher.cpp:
(WebCore::InspectorFrontendAPIDispatcher::frontendGlobalObject):
(WebCore::InspectorFrontendAPIDispatcher::evaluateExpression):
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::evaluateScriptInExtensionTab):
* Source/WebCore/inspector/PageDebugger.cpp:
(WebCore::PageDebugger::isContentScript const):
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::didClearWindowObjectInWorld):
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::didClearWindowObjectInWorld):
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
(WebCore::PageRuntimeAgent::reportExecutionContextCreation):
* Source/WebCore/loader/HistoryController.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::isWebChromeClient const):
* Source/WebCore/page/FocusController.h:
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/mediastream/WebRTCProvider.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
(isType):
* Source/WebCore/plugins/PluginViewBase.h:
(WebCore::PluginViewBase::isPluginView const):
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.h:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.h:
(isType):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDOMWindow.cpp:
(webkit_dom_dom_window_webkit_message_handlers_post_message):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleScriptWorld.cpp:
(WebKit::InjectedBundleScriptWorld::getOrCreate):
(WebKit::InjectedBundleScriptWorld::normalWorldSingleton):
(WebKit::InjectedBundleScriptWorld::protectedCoreWorld const):
(WebKit::InjectedBundleScriptWorld::protectedCoreWorld):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleScriptWorld.h:
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.cpp:
(WebKit::RemoteWebInspectorUI::windowObjectCleared):
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::windowObjectCleared):
* Source/WebKit/WebProcess/Plugins/PluginView.h:
(isType):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
(isType):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::platformInitialize):
(WebKit::WebPage::requestActiveNowPlayingSessionInfo):
(WebKit::WebPage::performDictionaryLookupAtLocation):
(WebKit::WebPage::dictionaryPopupInfoForRange):
(WebKit::WebPage::insertDictatedTextAsync):
(WebKit::WebPage::addDictationAlternative):
(WebKit::WebPage::dictationAlternativesAtSelection):
(WebKit::WebPage::clearDictationAlternatives):
(WebKit::WebPage::getContentsAsAttributedString):
(WebKit::WebPage::getPlatformEditorStateCommon const):
(WebKit::WebPage::getPDFFirstPageSize):
(WebKit::WebPage::replaceImageForRemoveBackground):
(WebKit::WebPage::readSelectionFromPasteboard):
(WebKit::elementHasHiddenVisibility):
(WebKit::WebPage::createTextIndicatorForElementWithID):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::protectedLocalFrameLoaderClient const):
(WebKit::WebFrame::jsContext):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::SendStopResponsivenessTimer::~SendStopResponsivenessTimer):
(WebKit::m_textAnimationController):
(WebKit::WebPage::requestMediaPlaybackState):
(WebKit::WebPage::pauseAllMediaPlayback):
(WebKit::WebPage::suspendAllMediaPlayback):
(WebKit::WebPage::resumeAllMediaPlayback):
(WebKit::WebPage::suspendAllMediaBuffering):
(WebKit::WebPage::resumeAllMediaBuffering):
(WebKit::WebPage::~WebPage):
(WebKit::WebPage::editorState const):
(WebKit::WebPage::shouldDispatchSyntheticMouseEventsWhenModifyingSelection const):
(WebKit::WebPage::renderTreeExternalRepresentation const):
(WebKit::WebPage::renderTreeExternalRepresentationForPrinting const):
(WebKit::WebPage::renderTreeSize const):
(WebKit::WebPage::pluginViewForFrame):
(WebKit::WebPage::executeEditingCommand):
(WebKit::WebPage::setEditable):
(WebKit::WebPage::setBaseWritingDirection):
(WebKit::WebPage::close):
(WebKit::WebPage::tryClose):
(WebKit::WebPage::suspendForProcessSwap):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::loadDataImpl):
(WebKit::WebPage::fromCorePage):
(WebKit::WebPage::setSize):
(WebKit::WebPage::didScalePage):
(WebKit::WebPage::setDeviceScaleFactor):
(WebKit::WebPage::screenPropertiesDidChange):
(WebKit::WebPage::listenForLayoutMilestones):
(WebKit::WebPage::setSuppressScrollbarAnimations):
(WebKit::WebPage::setEnableVerticalRubberBanding):
(WebKit::WebPage::setEnableHorizontalRubberBanding):
(WebKit::WebPage::setPaginationMode):
(WebKit::WebPage::setPaginationBehavesLikeColumns):
(WebKit::WebPage::setPageLength):
(WebKit::WebPage::setGapBetweenPages):
(WebKit::WebPage::postInjectedBundleMessage):
(WebKit::WebPage::setUnderPageBackgroundColorOverride):
(WebKit::WebPage::setHeaderPageBanner):
(WebKit::WebPage::setFooterPageBanner):
(WebKit::WebPage::setHeaderBannerHeight):
(WebKit::WebPage::setFooterBannerHeight):
(WebKit::snapshotColorSpace):
(WebKit::WebPage::pageStoppedScrolling):
(WebKit::WebPage::contextMenuAtPointInWindow):
(WebKit::WebPage::updateDrawingAreaLayerTreeFreezeState):
(WebKit::WebPage::mouseEvent):
(WebKit::WebPage::setLastKnownMousePosition):
(WebKit::WebPage::wheelEvent):
(WebKit::WebPage::keyEvent):
(WebKit::WebPage::validateCommand):
(WebKit::WebPage::requestFontAttributesAtSelectionStart):
(WebKit::WebPage::setObscuredContentInsets):
(WebKit::WebPage::setInitialFocus):
(WebKit::WebPage::setCanStartMediaTimerFired):
(WebKit::WebPage::updateIsInWindow):
(WebKit::WebPage::windowActivityDidChange):
(WebKit::WebPage::setActivityState):
(WebKit::WebPage::setNeedsDOMWindowResizeEvent):
(WebKit::WebPage::setUserAgent):
(WebKit::WebPage::suspendActiveDOMObjectsAndAnimations):
(WebKit::WebPage::resumeActiveDOMObjectsAndAnimations):
(WebKit::WebPage::suspend):
(WebKit::WebPage::resume):
(WebKit::WebPage::runJavaScript):
(WebKit::WebPage::runJavaScriptInFrameInScriptWorld):
(WebKit::frameWithSelection):
(WebKit::WebPage::getMainResourceDataOfFrame):
(WebKit::WebPage::preferencesDidChange):
(WebKit::WebPage::updatePreferences):
(WebKit::WebPage::layoutIfNeeded):
(WebKit::WebPage::updateRendering):
(WebKit::WebPage::shouldTriggerRenderingUpdate const):
(WebKit::WebPage::protectedPlaybackSessionManager):
(WebKit::WebPage::videoPresentationManager):
(WebKit::WebPage::videoControlsManagerDidChange):
(WebKit::WebPage::addConsoleMessage):
(WebKit::WebPage::notifyReportObservers):
(WebKit::WebPage::sendReportToEndpoints):
(WebKit::WebPage::performDragControllerAction):
(WebKit::WebPage::dragEnded):
(WebKit::WebPage::mayPerformUploadDragDestinationAction):
(WebKit::WebPage::unapplyEditCommand):
(WebKit::WebPage::reapplyEditCommand):
(WebKit::WebPage::voiceActivityDetected):
(WebKit::WebPage::changeSpellingToWord):
(WebKit::WebPage::unmarkAllMisspellings):
(WebKit::WebPage::unmarkAllBadGrammar):
(WebKit::WebPage::uppercaseWord):
(WebKit::WebPage::lowercaseWord):
(WebKit::WebPage::setTextForActivePopupMenu):
(WebKit::WebPage::replaceSelectionWithText):
(WebKit::WebPage::mainFrameDidLayout):
(WebKit::WebPage::dispatchMessage):
(WebKit::WebPage::scrollingCoordinator const):
(WebKit::WebPage::SandboxExtensionTracker::didFailProvisionalLoad):
(WebKit::WebPage::pdfDocumentForPrintingFrame):
(WebKit::WebPage::setUseColorAppearance):
(WebKit::WebPage::setMediaVolume):
(WebKit::WebPage::setMuted):
(WebKit::WebPage::stopMediaCapture):
(WebKit::pageContainsAnyHorizontalScrollbars):
(WebKit::WebPage::localMainFrame const):
(WebKit::WebPage::localTopDocument const):
(WebKit::WebPage::mainFrameView const):
(WebKit::WebPage::setCompositionAsync):
(WebKit::WebPage::setWritingSuggestion):
(WebKit::WebPage::setAlwaysShowsHorizontalScroller):
(WebKit::WebPage::setAlwaysShowsVerticalScroller):
(WebKit::WebPage::didCommitLoad):
(WebKit::WebPage::didSameDocumentNavigationForFrame):
(WebKit::WebPage::currentSelectionAsRange):
(WebKit::WebPage::scheduleFullEditorStateUpdate):
(WebKit::WebPage::updateWebsitePolicies):
(WebKit::WebPage::notifyUserScripts):
(WebKit::WebPage::getSamplingProfilerOutput):
(WebKit::WebPage::postMessageWithAsyncReply):
(WebKit::WebPage::postSynchronousMessageForTesting):
(WebKit::WebPage::setUserInterfaceLayoutDirection):
(WebKit::WebPage::registerURLSchemeHandler):
(WebKit::WebPage::setIsSuspended):
(WebKit::WebPage::requestStorageAccess):
(WebKit::WebPage::didLoadFromRegistrableDomain):
(WebKit::WebPage::requestAttachmentIcon):
(WebKit::WebPage::textAutoSizingAdjustmentTimerFired):
(WebKit::WebPage::setOverriddenMediaType):
(WebKit::WebPage::updateCORSDisablingPatterns):
(WebKit::WebPage::requestTextRecognition):
(WebKit::WebPage::startVisualTranslation):
(WebKit::WebPage::requestImageBitmap):
(WebKit::WebPage::setAppHighlightsVisibility):
(WebKit::WebPage::insertTextPlaceholder):
(WebKit::WebPage::removeTextPlaceholder):
(WebKit::WebPage::updateImageAnimationEnabled):
(WebKit::WebPage::pauseAllAnimations):
(WebKit::WebPage::playAllAnimations):
(WebKit::WebPage::useRedirectionForCurrentNavigation):
(WebKit::WebPage::remotePostMessage):
(WebKit::WebPage::updateLastNodeBeforeWritingSuggestions):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebUndoStep.h:
(WebKit::WebUndoStep::protectedStep const):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::protectedRemoteImageDecoderAVFManager):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKitLegacy/mac/DOM/DOMInternal.mm:
(-[WebScriptObject _initializeScriptDOMNodeImp]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
(WebFrameLoaderClient::dispatchDidClearWindowObjectInWorld):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _stringByEvaluatingJavaScriptFromString:forceUserGesture:]):
(-[WebFrame _stringByEvaluatingJavaScriptFromString:withGlobalObject:inScriptWorld:]):
(-[WebFrame globalContext]):
* Source/WebKitLegacy/mac/WebView/WebScriptWorld.mm:
(+[WebScriptWorld standardWorld]):
(+[WebScriptWorld findOrCreateWorld:]):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView aeDescByEvaluatingJavaScriptFromString:]):

Canonical link: <a href="https://commits.webkit.org/290504@main">https://commits.webkit.org/290504@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2af8d13f9350604f23aa8bd2e6b61faae2f12929

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90253 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9782 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/45176 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95256 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/41029 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/92305 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/10170 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/18099 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/69476 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/27075 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93254 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/7795 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/81877 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49836 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7520 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/36248 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/40160 "Hash 2af8d13f for PR 40678 does not build (failure)") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/83057 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/77849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/37307 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/97081 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/89031 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17441 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/12836 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/78476 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/17698 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77702 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77682 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/19185 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22144 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/20752 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/10700 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17451 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/22778 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/111522 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/17192 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/26688 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/20644 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/18976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

pvollan reviewed (approved):
R=me.

darinadler commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -5103,7 +5109,8 @@ bool WebPage::shouldTriggerRenderingUpdate(unsigned rescheduledRenderingUpdateCo
     static constexpr unsigned maxDelayedRenderingUpdateCount = 2;
-    if (m_remoteRenderingBackendProxy && m_remoteRenderingBackendProxy->delayedRenderingUpdateCount() > maxDelayedRenderingUpdateCount)
+    RefPtr remoteRenderingBackendProxy = m_remoteRenderingBackendProxy;
+    if (remoteRenderingBackendProxy && remoteRenderingBackendProxy->delayedRenderingUpdateCount() > maxDelayedRenderingUpdateCount)
```
```suggestion
    RefPtr proxy = m_remoteRenderingBackendProxy;
    if (proxy && proxy->delayedRenderingUpdateCount() > maxDelayedRenderingUpdateCount)
```

darinadler commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -6039,10 +6051,8 @@ void WebPage::capitalizeWord(FrameIdentifier frameID)
-
-    m_activePopupMenu->setTextForIndex(index);
+    if (RefPtr activePopupMenu = m_activePopupMenu)
+        activePopupMenu->setTextForIndex(index);
```
```suggestion
    if (RefPtr menu = m_activePopupMenu)
        menu->setTextForIndex(index);
```

darinadler commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -6343,11 +6353,8 @@ void WebPage::SandboxExtensionTracker::didFailProvisionalLoad(WebFrame* frame)
-    m_provisionalSandboxExtension->revoke();
-    m_provisionalSandboxExtension = nullptr;
+    if (RefPtr provisionalSandboxExtension = std::exchange(m_provisionalSandboxExtension, nullptr))
+        provisionalSandboxExtension->revoke();
```
```suggestion
    if (RefPtr extension = std::exchange(m_provisionalSandboxExtension, nullptr))
        extension->revoke();
```

darinadler commented on Source/WebKitLegacy/mac/DOM/DOMInternal.mm:
```diff
@@ -119,7 +119,7 @@ - (void)_initializeScriptDOMNodeImp
 
     // The global object which should own this node - FIXME: does this need to be isolated-world aware?
-    WebCore::JSDOMGlobalObject* globalObject = frame->script().globalObject(WebCore::mainThreadNormalWorld());
+    WebCore::JSDOMGlobalObject* globalObject = frame->script().globalObject(WebCore::mainThreadNormalWorldSingleton());
```
```suggestion
    auto* globalObject = frame->script().globalObject(WebCore::mainThreadNormalWorldSingleton());
```

darinadler commented on Source/WebKitLegacy/mac/WebView/WebFrame.mm:
```diff
@@ -703,7 +703,7 @@ - (NSString *)_stringByEvaluatingJavaScriptFromString:(NSString *)string forceUs
 #if PLATFORM(IOS_FAMILY)
     ASSERT(WebThreadIsLockedOrDisabled());
-    JSC::JSGlobalObject* lexicalGlobalObject = _private->coreFrame->script().globalObject(WebCore::mainThreadNormalWorld());
+    JSC::JSGlobalObject* lexicalGlobalObject = _private->coreFrame->script().globalObject(WebCore::mainThreadNormalWorldSingleton());
```
```suggestion
    auto* lexicalGlobalObject = _private->coreFrame->script().globalObject(WebCore::mainThreadNormalWorldSingleton());
```

darinadler commented on Source/WebKitLegacy/mac/WebView/WebFrame.mm:
```diff
@@ -719,7 +719,7 @@ - (NSString *)_stringByEvaluatingJavaScriptFromString:(NSString *)string forceUs
 
 #if !PLATFORM(IOS_FAMILY)
-    JSC::JSGlobalObject* lexicalGlobalObject = _private->coreFrame->script().globalObject(WebCore::mainThreadNormalWorld());
+    JSC::JSGlobalObject* lexicalGlobalObject = _private->coreFrame->script().globalObject(WebCore::mainThreadNormalWorldSingleton());
```
```suggestion
    auto* lexicalGlobalObject = _private->coreFrame->script().globalObject(WebCore::mainThreadNormalWorldSingleton());
```

cdumez committed:
Address safer C++ static analysis warnings in WebPage
https://bugs.webkit.org/show_bug.cgi?id=287760

Reviewed by Darin Adler and Per Arne Vollan.

* Source/WebCore/bindings/js/DOMWrapperWorld.cpp:
(WebCore::mainThreadNormalWorldSingleton):
(WebCore::mainThreadNormalWorld): Deleted.
* Source/WebCore/bindings/js/DOMWrapperWorld.h:
(WebCore::protectedMainThreadNormalWorld):
(WebCore::debuggerWorld):
(WebCore::pluginWorld):
* Source/WebCore/bindings/js/JSDOMWindowCustom.cpp:
(WebCore::mainWorldGlobalObject):
* Source/WebCore/bindings/js/JSLazyEventListener.cpp:
(WebCore::JSLazyEventListener::JSLazyEventListener):
(WebCore::JSLazyEventListener::create):
* Source/WebCore/bindings/js/ScriptController.cpp:
(WebCore::ScriptController::evaluateIgnoringException):
(WebCore::ScriptController::loadModuleScript):
(WebCore::ScriptController::linkAndEvaluateModuleScript):
(WebCore::ScriptController::evaluateModule):
(WebCore::ScriptController::setEvalEnabled):
(WebCore::ScriptController::setWebAssemblyEnabled):
(WebCore::ScriptController::setRequiresTrustedTypes):
(WebCore::ScriptController::executeScriptIgnoringException):
(WebCore::ScriptController::executeJavaScriptURL):
(WebCore::ScriptController::reportExceptionFromScriptError):
(WebCore::ScriptController::registerImportMap):
* Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
(WebCore::JSVMClientData::getAllWorlds):
* Source/WebCore/dom/Document.cpp:
(WebCore::Document::pictureInPictureElement const):
* Source/WebCore/dom/Element.cpp:
(WebCore::Element::setAttributeEventListener):
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::ScriptExecutionContext::globalObject const):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::ReplacementFragment::ReplacementFragment):
* Source/WebCore/html/HTMLBodyElement.cpp:
(WebCore::HTMLBodyElement::attributeChanged):
* Source/WebCore/html/HTMLFrameSetElement.cpp:
(WebCore::HTMLFrameSetElement::attributeChanged):
* Source/WebCore/inspector/InspectorController.cpp:
(WebCore::InspectorController::didClearWindowObjectInWorld):
* Source/WebCore/inspector/InspectorFrontendAPIDispatcher.cpp:
(WebCore::InspectorFrontendAPIDispatcher::frontendGlobalObject):
(WebCore::InspectorFrontendAPIDispatcher::evaluateExpression):
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::evaluateScriptInExtensionTab):
* Source/WebCore/inspector/PageDebugger.cpp:
(WebCore::PageDebugger::isContentScript const):
* Source/WebCore/inspector/agents/InspectorPageAgent.cpp:
(WebCore::InspectorPageAgent::didClearWindowObjectInWorld):
* Source/WebCore/inspector/agents/page/PageDebuggerAgent.cpp:
(WebCore::PageDebuggerAgent::didClearWindowObjectInWorld):
* Source/WebCore/inspector/agents/page/PageRuntimeAgent.cpp:
(WebCore::PageRuntimeAgent::reportExecutionContextCreation):
* Source/WebCore/loader/HistoryController.h:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::isWebChromeClient const):
* Source/WebCore/page/FocusController.h:
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/mediastream/WebRTCProvider.h:
* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProvider.h:
(isType):
* Source/WebCore/plugins/PluginViewBase.h:
(WebCore::PluginViewBase::isPluginView const):
* Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.h:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.h:
(isType):
* Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMDOMWindow.cpp:
(webkit_dom_dom_window_webkit_message_handlers_post_message):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleScriptWorld.cpp:
(WebKit::InjectedBundleScriptWorld::getOrCreate):
(WebKit::InjectedBundleScriptWorld::normalWorldSingleton):
(WebKit::InjectedBundleScriptWorld::protectedCoreWorld const):
(WebKit::InjectedBundleScriptWorld::protectedCoreWorld):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundleScriptWorld.h:
* Source/WebKit/WebProcess/Inspector/RemoteWebInspectorUI.cpp:
(WebKit::RemoteWebInspectorUI::windowObjectCleared):
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::windowObjectCleared):
* Source/WebKit/WebProcess/Plugins/PluginView.h:
(isType):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
(isType):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::platformInitialize):
(WebKit::WebPage::requestActiveNowPlayingSessionInfo):
(WebKit::WebPage::performDictionaryLookupAtLocation):
(WebKit::WebPage::dictionaryPopupInfoForRange):
(WebKit::WebPage::insertDictatedTextAsync):
(WebKit::WebPage::addDictationAlternative):
(WebKit::WebPage::dictationAlternativesAtSelection):
(WebKit::WebPage::clearDictationAlternatives):
(WebKit::WebPage::getContentsAsAttributedString):
(WebKit::WebPage::getPlatformEditorStateCommon const):
(WebKit::WebPage::getPDFFirstPageSize):
(WebKit::WebPage::replaceImageForRemoveBackground):
(WebKit::WebPage::readSelectionFromPasteboard):
(WebKit::elementHasHiddenVisibility):
(WebKit::WebPage::createTextIndicatorForElementWithID):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::protectedLocalFrameLoaderClient const):
(WebKit::WebFrame::jsContext):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::SendStopResponsivenessTimer::~SendStopResponsivenessTimer):
(WebKit::m_textAnimationController):
(WebKit::WebPage::requestMediaPlaybackState):
(WebKit::WebPage::pauseAllMediaPlayback):
(WebKit::WebPage::suspendAllMediaPlayback):
(WebKit::WebPage::resumeAllMediaPlayback):
(WebKit::WebPage::suspendAllMediaBuffering):
(WebKit::WebPage::resumeAllMediaBuffering):
(WebKit::WebPage::~WebPage):
(WebKit::WebPage::editorState const):
(WebKit::WebPage::shouldDispatchSyntheticMouseEventsWhenModifyingSelection const):
(WebKit::WebPage::renderTreeExternalRepresentation const):
(WebKit::WebPage::renderTreeExternalRepresentationForPrinting const):
(WebKit::WebPage::renderTreeSize const):
(WebKit::WebPage::pluginViewForFrame):
(WebKit::WebPage::executeEditingCommand):
(WebKit::WebPage::setEditable):
(WebKit::WebPage::setBaseWritingDirection):
(WebKit::WebPage::close):
(WebKit::WebPage::tryClose):
(WebKit::WebPage::suspendForProcessSwap):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::loadDataImpl):
(WebKit::WebPage::fromCorePage):
(WebKit::WebPage::setSize):
(WebKit::WebPage::didScalePage):
(WebKit::WebPage::setDeviceScaleFactor):
(WebKit::WebPage::screenPropertiesDidChange):
(WebKit::WebPage::listenForLayoutMilestones):
(WebKit::WebPage::setSuppressScrollbarAnimations):
(WebKit::WebPage::setEnableVerticalRubberBanding):
(WebKit::WebPage::setEnableHorizontalRubberBanding):
(WebKit::WebPage::setPaginationMode):
(WebKit::WebPage::setPaginationBehavesLikeColumns):
(WebKit::WebPage::setPageLength):
(WebKit::WebPage::setGapBetweenPages):
(WebKit::WebPage::postInjectedBundleMessage):
(WebKit::WebPage::setUnderPageBackgroundColorOverride):
(WebKit::WebPage::setHeaderPageBanner):
(WebKit::WebPage::setFooterPageBanner):
(WebKit::WebPage::setHeaderBannerHeight):
(WebKit::WebPage::setFooterBannerHeight):
(WebKit::snapshotColorSpace):
(WebKit::WebPage::pageStoppedScrolling):
(WebKit::WebPage::contextMenuAtPointInWindow):
(WebKit::WebPage::updateDrawingAreaLayerTreeFreezeState):
(WebKit::WebPage::mouseEvent):
(WebKit::WebPage::setLastKnownMousePosition):
(WebKit::WebPage::wheelEvent):
(WebKit::WebPage::keyEvent):
(WebKit::WebPage::validateCommand):
(WebKit::WebPage::requestFontAttributesAtSelectionStart):
(WebKit::WebPage::setObscuredContentInsets):
(WebKit::WebPage::setInitialFocus):
(WebKit::WebPage::setCanStartMediaTimerFired):
(WebKit::WebPage::updateIsInWindow):
(WebKit::WebPage::windowActivityDidChange):
(WebKit::WebPage::setActivityState):
(WebKit::WebPage::setNeedsDOMWindowResizeEvent):
(WebKit::WebPage::setUserAgent):
(WebKit::WebPage::suspendActiveDOMObjectsAndAnimations):
(WebKit::WebPage::resumeActiveDOMObjectsAndAnimations):
(WebKit::WebPage::suspend):
(WebKit::WebPage::resume):
(WebKit::WebPage::runJavaScript):
(WebKit::WebPage::runJavaScriptInFrameInScriptWorld):
(WebKit::frameWithSelection):
(WebKit::WebPage::getMainResourceDataOfFrame):
(WebKit::WebPage::preferencesDidChange):
(WebKit::WebPage::updatePreferences):
(WebKit::WebPage::layoutIfNeeded):
(WebKit::WebPage::updateRendering):
(WebKit::WebPage::shouldTriggerRenderingUpdate const):
(WebKit::WebPage::protectedPlaybackSessionManager):
(WebKit::WebPage::videoPresentationManager):
(WebKit::WebPage::videoControlsManagerDidChange):
(WebKit::WebPage::addConsoleMessage):
(WebKit::WebPage::notifyReportObservers):
(WebKit::WebPage::sendReportToEndpoints):
(WebKit::WebPage::performDragControllerAction):
(WebKit::WebPage::dragEnded):
(WebKit::WebPage::mayPerformUploadDragDestinationAction):
(WebKit::WebPage::unapplyEditCommand):
(WebKit::WebPage::reapplyEditCommand):
(WebKit::WebPage::voiceActivityDetected):
(WebKit::WebPage::changeSpellingToWord):
(WebKit::WebPage::unmarkAllMisspellings):
(WebKit::WebPage::unmarkAllBadGrammar):
(WebKit::WebPage::uppercaseWord):
(WebKit::WebPage::lowercaseWord):
(WebKit::WebPage::setTextForActivePopupMenu):
(WebKit::WebPage::replaceSelectionWithText):
(WebKit::WebPage::mainFrameDidLayout):
(WebKit::WebPage::dispatchMessage):
(WebKit::WebPage::scrollingCoordinator const):
(WebKit::WebPage::SandboxExtensionTracker::didFailProvisionalLoad):
(WebKit::WebPage::pdfDocumentForPrintingFrame):
(WebKit::WebPage::setUseColorAppearance):
(WebKit::WebPage::setMediaVolume):
(WebKit::WebPage::setMuted):
(WebKit::WebPage::stopMediaCapture):
(WebKit::pageContainsAnyHorizontalScrollbars):
(WebKit::WebPage::localMainFrame const):
(WebKit::WebPage::localTopDocument const):
(WebKit::WebPage::mainFrameView const):
(WebKit::WebPage::setCompositionAsync):
(WebKit::WebPage::setWritingSuggestion):
(WebKit::WebPage::setAlwaysShowsHorizontalScroller):
(WebKit::WebPage::setAlwaysShowsVerticalScroller):
(WebKit::WebPage::didCommitLoad):
(WebKit::WebPage::didSameDocumentNavigationForFrame):
(WebKit::WebPage::currentSelectionAsRange):
(WebKit::WebPage::scheduleFullEditorStateUpdate):
(WebKit::WebPage::updateWebsitePolicies):
(WebKit::WebPage::notifyUserScripts):
(WebKit::WebPage::getSamplingProfilerOutput):
(WebKit::WebPage::postMessageWithAsyncReply):
(WebKit::WebPage::postSynchronousMessageForTesting):
(WebKit::WebPage::setUserInterfaceLayoutDirection):
(WebKit::WebPage::registerURLSchemeHandler):
(WebKit::WebPage::setIsSuspended):
(WebKit::WebPage::requestStorageAccess):
(WebKit::WebPage::didLoadFromRegistrableDomain):
(WebKit::WebPage::requestAttachmentIcon):
(WebKit::WebPage::textAutoSizingAdjustmentTimerFired):
(WebKit::WebPage::setOverriddenMediaType):
(WebKit::WebPage::updateCORSDisablingPatterns):
(WebKit::WebPage::requestTextRecognition):
(WebKit::WebPage::startVisualTranslation):
(WebKit::WebPage::requestImageBitmap):
(WebKit::WebPage::setAppHighlightsVisibility):
(WebKit::WebPage::insertTextPlaceholder):
(WebKit::WebPage::removeTextPlaceholder):
(WebKit::WebPage::updateImageAnimationEnabled):
(WebKit::WebPage::pauseAllAnimations):
(WebKit::WebPage::playAllAnimations):
(WebKit::WebPage::useRedirectionForCurrentNavigation):
(WebKit::WebPage::remotePostMessage):
(WebKit::WebPage::updateLastNodeBeforeWritingSuggestions):
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebUndoStep.h:
(WebKit::WebUndoStep::protectedStep const):
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::protectedRemoteImageDecoderAVFManager):
* Source/WebKit/WebProcess/WebProcess.h:
* Source/WebKitLegacy/mac/DOM/DOMInternal.mm:
(-[WebScriptObject _initializeScriptDOMNodeImp]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebFrameLoaderClient.mm:
(WebFrameLoaderClient::dispatchDidClearWindowObjectInWorld):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _stringByEvaluatingJavaScriptFromString:forceUserGesture:]):
(-[WebFrame _stringByEvaluatingJavaScriptFromString:withGlobalObject:inScriptWorld:]):
(-[WebFrame globalContext]):
* Source/WebKitLegacy/mac/WebView/WebScriptWorld.mm:
(+[WebScriptWorld standardWorld]):
(+[WebScriptWorld findOrCreateWorld:]):
* Source/WebKitLegacy/mac/WebView/WebView.mm:
(-[WebView aeDescByEvaluatingJavaScriptFromString:]):

Canonical link: https://commits.webkit.org/290504@main
